### PR TITLE
Fix asset precompile in docker build workflow

### PR DESCRIPTION
**Pull request by djpremier**
Complementing the work started in PR #737 so that the tests run successfully.

Solve Issue https://github.com/globalize/globalize/issues/723

### CODE REVIEW

**Comment by djpremier** [comment]
In the first "execution" of the `connected?` function it can happen that Rails does not identify an active connection and with that the result is negative (false), using the function together with the `connection` makes it possible that in the current scope there is no connection, the search function exists globally. Here's a reference:

https://api.rubyonrails.org/classes/ActiveRecord/ConnectionHandling.html#method-i-connection

**Comment by djpremier** [comment]
Awaiting test workflow approval, but in local all is good!

image

And also in Azure CI

image

**Comment by djpremier** [comment]
@parndt Can you review this? 🙂

File: test/support/database.yml [hunk 1]
```diff
@@ -9,10 +9,11 @@ mysql:
 
 postgresql:
   adapter: postgresql
-  host: localhost
-  port: 5432
+  host: <%= ENV.fetch('POSTGRESQL_HOST', 'localhost') %>
+  port: <%= ENV.fetch('POSTGRESQL_PORT', 5432) %>
   database: globalize_test
   username: postgres
+  password: <%= ENV['POSTGRESQL_PASSWORD'] %>
```
Comment (parndt) [inline test/support/database.yml:16]:
```suggestion
  host: <%= ENV.fetch("PGHOST", "localhost") %>
  port: <%= ENV.fetch("PGPORT", 5432) %>
  database: globalize_test
  username: postgres
<% if ENV["PGPASSWORD"] -%>
  password: <%= ENV["PGPASSWORD"] %>
<% end %>
```

Maybe we can use these vars which are documented in the postgresql docs as well: https://www.postgresql.org/docs/current/libpq-envars.html

Comment (djpremier) [reply test/support/database.yml]: Hi @parndt, thanks for answer, updated!

**Comment by parndt** [comment]
Looks decent @djpremier; we now have duplication in the code which we could solve by extracting a private method like this:

```ruby
def database_connection_possible?
  begin
    # Without a connection tentative, the `connected?` function can responds with a false negative
    ::ActiveRecord::Base.connection
  rescue
    # Ignore connection fail because in docker build hasn't a database connection
    nil
  end

  ::ActiveRecord::Base.connected? && 
    table_exists? && 
    translation_class.table_exists?
end
```

which we then use like this:

```ruby
if Globalize.rails_5? && database_connection_possible?
```

and this:
```ruby
return unless Globalize.rails_42? && database_connection_possible?
```

What do you think?

**Comment by djpremier** [comment]
> Looks decent @djpremier; we now have duplication in the code which we could solve by extracting a private method like this:
> 
> ```ruby
> def database_connection_possible?
>   begin
>     # Without a connection tentative, the `connected?` function can responds with a false negative
>     ::ActiveRecord::Base.connection
>   rescue
>     # Ignore connection fail because in docker build hasn't a database connection
>     nil
>   end
> 
>   ::ActiveRecord::Base.connected? && 
>     table_exists? && 
>     translation_class.table_exists?
> end
> ```
> 
> which we then use like this:
> 
> ```ruby
> if Globalize.rails_5? && database_connection_possible?
> ```
> 
> and this:
> 
> ```ruby
> return unless Globalize.rails_42? && database_connection_possible?
> ```
> 
> What do you think?

Hello, uhm I think it's interesting, it might be useful in the future in a possible expansion of features, I'll change it here.

**Review by parndt — APPROVED** [review]

**Comment by parndt** [comment]
I've bumped the version, added this change to the change log, and will release it as `6.2.0` thank you 😄

**Pull request merged**